core/http-auth/src/main/java/software/amazon/awssdk/http/auth/internal/DefaultAwsV4HttpSigner.java

```diff
@@ -52,16 +57,33 @@
 import software.amazon.awssdk.utils.CompletableFutureUtils;
 import software.amazon.awssdk.utils.Logger;
 import software.amazon.awssdk.utils.StringUtils;
-import software.amazon.awssdk.utils.http.SdkHttpUtils;
 
 /**
  * A default implementation of {@link AwsV4HttpSigner}.
  */
 @SdkInternalApi
-public final class DefaultAwsV4HttpSigner implements AwsV4HttpSigner {
+public class DefaultAwsV4HttpSigner implements AwsV4HttpSigner {
 
     private static final Logger LOG = Logger.loggerFor(DefaultAwsV4HttpSigner.class);
 
+    // required
+    protected String regionName;
```

Comment:
This makes the signers stateful and not usable across parallel requests. I know the existing code `AuthScheme.signer()` is creating a new instance of signer, but I had been considering a change to use a single instance. I'd want to double check with @millems if there was any requirement of the signers to be thread safe.

Reply:
Yep. If we don't want this, then we'll need to refactor all of this to a "parameters" object. The `setParameters` method would then create this object, and every method would pass it along. The signer then becomes re-usable, but that parameters object has to be created on every request.